app/models/authentication.rb

```diff
@@ -19,6 +19,10 @@ class Authentication < ApplicationRecord
   private
 
   def set_source
-    self.source_id = resource&.source_id
+    self.source_id = if resource.instance_of?(Source)
+                       resource.id
```

Comment:
@lindgrenj6 when this happens?

Reply:
btw using `resource_type` and `resource_id` saves one DB request

Reply:
This happens when using the `Source` as the `resource` for an Authentication. Which I hope is what we're going to be doing for superkey authentications that way we don't have to have a dummy endpoint or anything like that involved.